compiler/literate/html.ml

```diff
@@ -185,45 +186,67 @@ let rec law_structure_to_html
   | A.LawHeading (heading, children) ->
     let h_number = heading.law_heading_precedence + 1 in
     let h_name = Marked.unmark heading.law_heading_name in
+    let complete_headings = parents_headings @ [h_name] in
+    let id = complete_headings |> String.concat "-" |> sanitize_html_href in
     let fmt_details_open fmt () =
       if 2 = h_number then
         Format.fprintf fmt "<details><summary>%s</summary>" h_name
     in
     let fmt_details_close fmt () =
-      if 2 >= h_number then Format.fprintf fmt "</details>"
+      if 2 = h_number then Format.fprintf fmt "</details>"
```

Comment:
So with that, you only enclose items at level 2 of sectionning or below ? It sounds reasonable to me, but there should be a comment detailing the logic of what's being done here.

Reply:
Yes, I think this is the best trade-off.